docs(rosetta): add Rosetta documentation page (#9688)
cosmos/cosmos-sdk: docs/rosetta/README.md

<!--
order: false
parent:
  order: 2
-->

# Rosetta

This folder contains instructions about how to use the Coinbase Rosetta API integration.

## Motivations and design

For the motivation an design choices refer to the [ADR 035](../architecture/adr-035-rosetta-api-support.md)

## Usage

The Rosetta API server is a stand-alone server that connects to a node of a chain developed with the Cosmos-SDK. 

To enable the Rosetta API server support it's required to Add the `RosettaCommand` to your application root command file.

> After the 
> 
> ```
> initRootCmd(rootCmd, encodingConfig)
> ```
> 
> line, add the following lines:

Reviewer: ````suggestion
Find the following line within the root command file:

```
initRootCmd(rootCmd, encodingConfig)
```

After that line, add the following:
````

```
rootCmd.AddCommand(
  server.RosettaCommand(encodingConfig.InterfaceRegistry, encodingConfig.Marshaler)
)
```

> The application root command file is in the server package in `github.com/cosmos/cosmos-sdk/server`.

Reviewer: ```suggestion
The `RosettaCommand` function builds the `rosetta` root command and is defined in the `server` package within Cosmos SDK.
```

> Since we’ve updated the Cosmos SDK to work with the Rosetta API, updating the application root command file is all you need to do.

Reviewer: ```suggestion
Since we’ve updated the Cosmos SDK to work with the Rosetta API, updating the application's root command file is all you need to do.
```
